ray_lightning/ray_ddp.py

```diff
@@ -274,87 +311,140 @@ def shutdown_remote():
 
     # All methods below are only executed in remote Ray workers.
 
+    def set_world_ranks(self, process_idx: int = 0):
+        """Set the appropriate rank attributes for the trainer."""
+        if self._is_remote:
```

Comment:
It should be a no-op. I'll add a comment about this. The reason that this is needed is because `DDPSpawnPlugin` calls `set_world_ranks` during `__init__` because it assumes in a multi-node setup the correct address and port environment variables have already been set by the user before running the script. But in our case, we only want to call this method after the workers have been initialized.